src/modules/vendor/pages/ni/pedagogy/QuestionnairesDirectory.vue

```diff
@@ -1,17 +1,100 @@
+          class="q-my-md q-mr-md" />
+      </div>
+      <div v-else class="text-italic">Aucun questionnaire "{{ questionnairesByType.type.label }}"</div>
+    </div>
+    <q-btn class="fixed fab-custom" no-caps rounded color="primary" icon="add" label="Créer un questionnaire"
+      @click="questionnaireCreationModal = true" :disable="loading" />
+
+    <questionnaire-creation-modal v-model="questionnaireCreationModal" @hide="resetCreationModal"
+      :loading="modalLoading" @submit="createQuestionnaire" :validations="$v.newQuestionnaire"
+      :new-questionnaire.sync="newQuestionnaire" />
   </q-page>
 </template>
 
 <script>
+import { required } from 'vuelidate/lib/validators';
+import groupBy from 'lodash/groupBy';
+import Questionnaires from '@api/Questionnaires';
 import TitleHeader from '@components/TitleHeader';
+import QuestionnaireCell from '@components/courses/QuestionnaireCell';
+import QuestionnaireCreationModal from 'src/modules/vendor/components/programs/QuestionnaireCreationModal';
+import { NotifyNegative, NotifyPositive, NotifyWarning } from '@components/popup/notify';
+import { QUESTIONNAIRE_TYPES } from '@data/constants';
 
 export default {
   metaInfo: { title: 'Questionnaires' },
   name: 'QuestionnairesDirectory',
   components: {
     'ni-title-header': TitleHeader,
+    'questionnaire-creation-modal': QuestionnaireCreationModal,
+    'questionnaire-cell': QuestionnaireCell,
+  },
+  data () {
+    return {
+      loading: false,
+      questionnaires: [],
+      modalLoading: false,
+      questionnaireCreationModal: false,
+      newQuestionnaire: { title: '', type: '' },
+      questionnaireTypes: [],
+    };
+  },
+  validations () {
+    return {
+      newQuestionnaire: { title: { required }, type: { required } },
+    };
+  },
+  async created () {
+    await this.refreshQuestionnaires();
+  },
+  methods: {
+    async refreshQuestionnaires () {
+      try {
+        this.loading = true;
+        const questionnaires = (await Questionnaires.list())
+          .sort((a, b) => new Date(b.createdAt) - new Date(a.createdAt));
+
+        const questionnairesByType = groupBy(questionnaires, q => q.type);
+
+        this.questionnaires = QUESTIONNAIRE_TYPES
+          .map(q => ({ type: q, questionnairesList: questionnairesByType[q.value] }));
+      } catch (e) {
+        console.error(e);
+        NotifyNegative('Erreur lors de la récupération des questionnaires.');
+      } finally {
+        this.loading = false;
+      }
+    },
+    resetCreationModal () {
+      this.$v.newQuestionnaire.$reset();
+      this.newQuestionnaire = { title: '' };
+    },
+    async createQuestionnaire () {
+      try {
+        this.$v.newQuestionnaire.$touch();
+        if (this.$v.newQuestionnaire.$error) return NotifyWarning('Champ(s) invalide(s)');
+
+        this.modalLoading = true;
+        await Questionnaires.create({ title: this.newQuestionnaire.title, type: this.newQuestionnaire.type });
```

Comment:
pourquoi tu recrées l'objet ? tu pourrais mettre directement `await Questionnaires.create(this.newQuestionnaire);` non ?

Reply:
done